pykokkos/interface/type_inference.py

```diff
@@ -0,0 +1,279 @@
+import inspect
+from dataclasses import dataclass
+from typing import  Callable, Dict, Optional, Tuple, Union, List, Any
+import pykokkos.kokkos_manager as km
+from .execution_policy import MDRangePolicy, TeamPolicy, TeamThreadRange, RangePolicy, ExecutionPolicy
+from .views import View, ViewType
+from .layout import Layout
+from .data_types import DataType, DataTypeClass
+
+@dataclass
+class HandledArgs:
+    """
+    Class for holding the arguments passed to parallel_* functions
+    """
+
+    name: Optional[str]
+    policy: ExecutionPolicy
+    workunit: Callable
+    view: Optional[ViewType]
+    initial_value: Union[int, float]
+
+
+@dataclass
+class UpdatedTypes:
+    """
+    Class for storing inferred type annotation information 
+    (Making Pykokkos more pythonic by automatically inferring types)
+    """
+
+    workunit: Callable
+    inferred_types: Dict[str, str] # type information stored as string: identifier -> type
+    is_arg: set[str]
+    layout_change: Dict[str, str] # if layout for a view is not Layout.Default it will be stored here
+
+
+def handle_args(is_for: bool, *args) -> HandledArgs:
+    """
+    Handle the *args passed to parallel_* functions
```

Comment:
Handle the *args passed to parallel_* functions =>
Handle the *args passed to functions

at this point it is not relevant any longer it is from parallel_ function; could be any in theory

Reply:
This method is being moved to type_inference from parallel_dispatch so we have everything argument related in one place. (not authored by me) But it does strictly deal with only parallel_<> functions. Which is why the specification is necessary

I think we should leave it as it is until we remove parallel_for etc.